code/modules/mob/living/basic/guardian/guardian_types/gaseous.dm

```diff
@@ -0,0 +1,153 @@
+/// Not particularly resistant, but versatile due to the selection of gases it can generate.
+/mob/living/basic/guardian/gaseous
+	guardian_type = GUARDIAN_GASEOUS
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 0)
+	range = 7
+	playstyle_string = span_holoparasite("As a <b>gaseous</b> type, you have only light damage resistance, but you can expel gas in an area. In addition, your punches cause sparks, and you make your summoner inflammable.")
+	creator_name = "Gaseous"
+	creator_desc = "Creates sparks on touch and continuously expels a gas of its choice. Automatically extinguishes the user if they catch on fire."
+	creator_icon = "gaseous"
+	toggle_button_type = /atom/movable/screen/guardian/toggle_mode/gases
+	/// Ability we use to select gases
+	var/datum/action/cooldown/mob_cooldown/expel_gas/gas
+	/// Rate of temperature stabilization per second.
+	var/temp_stabilization_rate = 0.1
+
+/mob/living/basic/guardian/gaseous/Initialize(mapload, theme)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_PRE_PRESSURE_PUSH, PROC_REF(pre_pressure_moved))
+	gas = new(src)
+	gas.owner_has_control = FALSE // It's nicely integrated with the Guardian UI, no need to have two buttons
+	gas.Grant(src)
+
+/mob/living/basic/guardian/gaseous/Destroy()
+	QDEL_NULL(gas)
+	return ..()
+
+/mob/living/basic/guardian/gaseous/toggle_modes()
+	gas.Trigger()
+
+/mob/living/basic/guardian/gaseous/Life(seconds_per_tick, times_fired)
```

Comment:
i feel like this might be better off as a signal since it does zero work on `src`